aten/src/ATen/native/LossMultiMargin.cpp

```diff
@@ -0,0 +1,337 @@
+  } else {
+    nframe = input.size(0);
+    dim = input.size(1);
+  }
+
+  TORCH_CHECK(
+      target.numel() > 0 && target.dim() <= 1 && target.numel() == nframe,
+      "inconsistent target size, got: ",
+      target.sizes());
+
+  grad_input.resize_as_(input);
+  TORCH_CHECK(grad_input.is_contiguous(), "grad_input must be contiguous");
+
+  auto input_contiguous = input.contiguous();
+  auto target_contiguous = target.contiguous();
+  auto weight_contiguous = weight.contiguous();
+  AT_DISPATCH_FLOATING_TYPES(
+      input.scalar_type(), "multi_margin_loss_backward_cpu_kernel", [&] {
+        auto grad_input_data = grad_input.data_ptr<scalar_t>();
+        auto input_data = input_contiguous.data_ptr<scalar_t>();
+        auto target_data = target_contiguous.data_ptr<int64_t>();
+        auto weight_data = weight_contiguous.defined()
+            ? weight_contiguous.data_ptr<scalar_t>()
+            : nullptr;
+        scalar_t g = reduction == Reduction::Mean
+            ? static_cast<scalar_t>(1. / (nframe * dim))
+            : static_cast<scalar_t>(1. / dim);
+        multi_margin_loss_backward_cpu_kernel<scalar_t>(
+            grad_input_data,
+            grad_output,
+            input_data,
+            target_data,
+            p,
+            margin.to<scalar_t>(),
+            g,
+            weight_data,
+            nframe,
+            dim,
+            reduction);
+      });
+}
+
+} // namespace
+
+Tensor multi_margin_loss_cpu(
+    const Tensor& input,
+    const Tensor& target,
+    Scalar p,
+    Scalar margin,
+    const Tensor& weight,
+    int64_t reduction) {
+  auto output = at::empty({0}, input.options());
+  multi_margin_loss_out_cpu_template(
+      output, input, target, p.toInt(), margin, weight, reduction);
+  return output;
+}
+
+Tensor& multi_margin_loss_cpu_out(
+    Tensor& output,
+    const Tensor& input,
+    const Tensor& target,
+    Scalar p,
+    Scalar margin,
+    const Tensor& weight,
+    int64_t reduction) {
+  multi_margin_loss_out_cpu_template(
+      output, input, target, p.toInt(), margin, weight, reduction);
+  return output;
+}
+
+Tensor multi_margin_loss_cpu_backward(
+    const Tensor& grad_output,
+    const Tensor& input,
+    const Tensor& target,
+    Scalar p,
+    Scalar margin,
+    const Tensor& weight,
+    int64_t reduction) {
+  auto grad_input = at::empty({0}, input.options());
+  multi_margin_loss_backward_out_cpu_template(
+      grad_input,
+      grad_output,
+      input,
+      target,
+      p.toInt(),
+      margin,
+      weight,
+      reduction);
+  return grad_input;
+}
+
+Tensor& multi_margin_loss_cpu_backward_out(
+    Tensor& grad_input,
+    const Tensor& grad_output,
+    const Tensor& input,
+    const Tensor& target,
+    Scalar p,
+    Scalar margin,
+    const Tensor& weight,
+    int64_t reduction) {
+  multi_margin_loss_backward_out_cpu_template(
+      grad_input,
+      grad_output,
+      input,
+      target,
+      p.toInt(),
```

Comment:
This is not anything you have to do. But I do note it's a bit weird that we take a Scalar as argument p, but then immediately turn around and say it must be an int.

Reply:
Actually this results from the fact how this function is defined in native_functions.yaml, where it is: `- func: multi_margin_loss(Tensor self, Tensor target, Scalar p=1, Scalar margin=1, Tensor? weight=None, int reduction=Mean) -> Tensor`. I did not touch this signature - that means the input will be a Scalar. Bu if you look at the documentation and in the python source there is the strange restriction for p to be either 1 or 2 and no other values are allowed. It would be relatively easy to support all power-values - maybe with special fast-cases for 1 and 2. I showed this to @VitalyFedyunin also. For this port I kept the original behavior -> the only thing I added was a check for p==1||p==2 which before was only happening in python, see: https://github.com/pytorch/pytorch/blob/951dd03037d01a4f9ad9703e4f0908e11c59f8c9/torch/nn/functional.py#L2323-L2324